### PR TITLE
throw exception Azure::Core::RequestFailedException or those inherit from that

### PR DESCRIPTION
**Pull request by Jinming-Hu**
closes https://github.com/Azure/azure-sdk-for-cpp/issues/1139

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp [hunk 1]
```diff
@@ -374,7 +374,14 @@ namespace Azure { namespace Storage { namespace Blobs {
     /**
      * @brief Optional conditions that must be met to perform this operation.
      */
-    BlobContainerAccessConditions AccessConditions;
+    struct : public LeaseAccessConditions
```
Comment (katmsft) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:377]: Not single purpose, and please add change log.

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:377]: It is single purpose. The main purpose of this PR is to correct storage exception types. In the past, we throw exception on unsupported access condition. Now that we removed the unsupported access condition, we don't need to throw any more

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:377]: Plus I don't think this need to be reflected in changelog.

Comment (katmsft) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:377]: This option used to have `IfModifiedSince` and `IfUnmodifiedSince`, but now only `IfModifiedSince` right? Why is that not something needs to be reflected in changelog?

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:377]: `IfUnmodifiedSince` never works. It throws in the past, now it crashes. Why would anybody care about it?

Comment (katmsft) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:377]: No, it does not crash now, it is a compiler error now. Are you saying that the precondition check changed to compiler error is not breaking change?

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:377]: I may be able to accept adding it into changelog? But is it a breaking change? I don't think so.

Comment (katmsft) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:377]: I believe it is either a breaking change, or we shouldn't mention it at all.
Consider if the customer assigns a `BlobContainerAccessConditions` everywhere in the options, it used to work if `IfUnmodifiedSince` is null, but not now for this API.

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:377]: well, okay

**Review by katmsft — APPROVED** [review]
The only concern is if the change to the option is a breaking change. Leave it to the author to decide.

**Pull request merged**